pygmt/base_plotting.py

```diff
@@ -344,8 +379,82 @@ def grdimage(self, grid, **kwargs):
+            *out_img* to select the image file name and extension. If the
+            extension is one of .bmp, .gif, .jpg, .png, or .tif then no driver
+            information is required. For other output formats you must append
+            the required GDAL driver. The *driver* is the driver code name used
+            by GDAL; see your GDAL installation's documentation for available
+            drivers. Append a **+c**\\ *options* string where options is a list
+            of one or more concatenated number of GDAL **-co** options. For
+            example, to write a GeoPDF with the TerraGo format use
+            ``=PDF+cGEO_ENCODING=OGC_BP``. Notes: (1) If a tiff file (.tif) is
+            selected then we will write a GeoTiff image if the GMT projection
+            syntax translates into a PROJ syntax, otherwise a plain tiff file
+            is produced. (2) Any vector elements will be lost.
+        {B}
+        {CPT}
+        img_in : str
+            ``[r]``
+            GMT will automatically detect standard image files (Geotiff, TIFF,
+            JPG, PNG, GIF, etc.) and will read those via GDAL. For very obscure
+            image formats you may need to explicitly set *img_in*, which
+            specifies that the grid is in fact an image file to be read via
+            GDAL. Append **r** to assign the region specified by *region*
+            to the image. For example, if you have used ``region='d'`` then the
+            image will be assigned a global domain. This mode allows you to
+            project a raw image (an image without referencing coordinates).
+        dpi : int
+            ``[i|dpi]``.
+            Sets the resolution of the projected grid that will be created if a
+            map projection other than Linear or Mercator was selected [100]. By
+            default, the projected grid will be of the same size (rows and
+            columns) as the input file. Specify **i** to use the PostScript
+            image operator to interpolate the image at the device resolution.
+        bit_color : str
+            ``color[+b|f]``.
+            This option only applies when a resulting 1-bit image otherwise
+            would consist of only two colors: black (0) and white (255). If so,
+            this option will instead use the image as a transparent mask and
+            paint the mask with the given color. Append **+b** to paint the
+            background pixels (1) or **+f** for the foreground pixels
+            [Default].
+        shading : str
+            ``[intensfile|intensity|modifiers]``.
+            Give the name of a grid file with intensities in the (-1,+1) range,
+            or a constant intensity to apply everywhere (affects the ambient
+            light). Alternatively, derive an intensity grid from the input data
+            grid via a call to `grdgradient`; append **+a** \\ *azimuth*,
+            **+n** \\ *args*, and **+m** \\ *ambient* to specify azimuth,
```

Comment:
There should be no spaces between modifier and value:
![image](https://user-images.githubusercontent.com/3974108/93945269-9a618400-fd04-11ea-9681-2127085ac12f.png)

```suggestion
            grid via a call to `grdgradient`; append **+a**\\ *azimuth*,
            **+n**\\ *args*, and **+m**\\ *ambient* to specify azimuth,
```

Are you sure we need double back-slashes?

Reply:
Good catch. Single backslash would result in a flake8 error.

Reply:
> Single backslash would result in a flake8 error `W605 invalid escape sequence '\ '`

Double backslashes in ReST syntax doesn't give what we want if they're not in docstring: http://rst.ninjs.org/#KiorYSoqXFwgKmF6aW11dGgq

Why does flake8 check ReST syntax? Is it a flake8 bug? Or can we make flake8 ignore it for ReST strings?

Reply:
Anyway, the documentation looks good now. We may see if we can avoid double backslashes in the future.

Reply:
> Double backslashes in ReST syntax doesn't give what we want if they're not in docstring: http://rst.ninjs.org/#KiorYSoqXFwgKmF6aW11dGgq

The double backslash is needed for Python, the alternative would be to use a raw string (e.g. `r"**+n**\ *args*). We came across this at https://github.com/GenericMappingTools/pygmt/pull/525#discussion_r454414687, might be good if we can standardize this in our CONTRIBUTING.md document (open an issue?).

Anyways, the renedered version looks fine:

![image](https://user-images.githubusercontent.com/23487320/93947007-cd505a00-fd8e-11ea-9977-17498818bccf.png)

> Why does flake8 check ReST syntax? Is it a flake8 bug? Or can we make flake8 ignore it for ReST strings?

A quick search yields https://github.com/kataev/flake8-rst, but I haven't tried it, and don't know if it does what we want.

Reply:
GMT uses singe backslash syntax like `**+a**\ *azimuth*`. If we copy documentation from GMT, then I expect we will see the same warning often.

Reply:
I'll open up an issue for this to continue the discussion :)